weasyl/controllers/moderation.py

```diff
@@ -49,58 +57,75 @@ def modcontrol_report_(request):
 
 @moderator_only
 def modcontrol_reports_(request):
-    form = request.web_input(status="open", violation="", submitter="")
+    form = {
+               "status": request.params.get('status', 'open'),
+               "violation": request.params.get('violation'),
+               "submitter": request.params.get('submitter', '')
+           }
     return Response(define.webpage(request.userid, "modcontrol/reports.html", [
         # Method
-        {"status": form.status, "violation": int(form.violation or -1), "submitter": form.submitter},
+        form,
         # Reports
-        report.select_list(request.userid, form),
+        report.select_list(**form),
         macro.MACRO_REPORT_VIOLATION,
     ], title="Reported Content"))
 
 
 @moderator_only
 @token_checked
 def modcontrol_closereport_(request):
-    form = request.web_input(reportid='', action='')
-    report.close(request.userid, form)
-    raise HTTPSeeOther(location="/modcontrol/report?reportid=%d" % (int(form.reportid),))
+    report.close(
+        request.userid,
+        request.params.get('reportid', ''),
+        request.params.get('action', ''),
+        request.params.get('explanation', ''),
+        request.params.get('note_title', ''),
+        request.params.get('user_note', ''),
+        'assign' in request.params,
+        'unassign' in request.params,
+        'close_all_user_reports' in request.params
+    )
+    raise HTTPSeeOther(location="/modcontrol/report?reportid=%d" % (int(request.params.get('reportid', '')),))
```

Comment:
`int("")` will throw:

```suggestion
    raise HTTPSeeOther(location="/modcontrol/report?reportid=%d" % (int(request.params['reportid']),))
```